x/nameservice/client/testutil/tx.go

```diff
@@ -0,0 +1,910 @@
+)
+
+var (
+	accountName    = "accountName"
+	accountAddress string
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	cfg     network.Config
+	network *network.Network
+	bondId  string
+}
+
+func NewIntegrationTestSuite(cfg network.Config) *IntegrationTestSuite {
+	return &IntegrationTestSuite{cfg: cfg}
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	s.T().Log("setting up integration test suite")
+
+	var genesisState = s.cfg.GenesisState
+	var nsData nstypes.GenesisState
+	s.Require().NoError(s.cfg.Codec.UnmarshalJSON(genesisState[nstypes.ModuleName], &nsData))
+
+	nsData.Params.RecordRent = sdk.NewCoin(s.cfg.BondDenom, nstypes.DefaultRecordRent)
+	nsData.Params.RecordRentDuration = 5 * time.Second
+	nsData.Params.AuthorityGracePeriod = 5 * time.Second
+	nsDataBz, err := s.cfg.Codec.MarshalJSON(&nsData)
+	s.Require().NoError(err)
+	genesisState[nstypes.ModuleName] = nsDataBz
+	s.cfg.GenesisState = genesisState
+
+	s.network = network.New(s.T(), s.cfg)
+
+	_, err = s.network.WaitForHeight(2)
+	s.Require().NoError(err)
+
+	// setting up random account
+	s.createAccountWithBalance(accountName)
+	CreateBond(s)
+	s.bondId = GetBondId(s)
+}
+
+func (s *IntegrationTestSuite) createAccountWithBalance(accountName string) {
+	val := s.network.Validators[0]
+	sr := s.Require()
+	consPrivKey := ed25519.GenPrivKey()
+	consPubKeyBz, err := s.cfg.Codec.MarshalInterfaceJSON(consPrivKey.PubKey())
+	sr.NoError(err)
+	sr.NotNil(consPubKeyBz)
+
+	info, _, err := val.ClientCtx.Keyring.NewMnemonic(accountName, keyring.English, sdk.FullFundraiserPath, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+	sr.NoError(err)
+
+	newAddr := sdk.AccAddress(info.GetPubKey().Address())
+	_, err = banktestutil.MsgSendExec(
+		val.ClientCtx,
+		val.Address,
+		newAddr,
+		sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(1000000000000000000))),
+		fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+	)
+	sr.NoError(err)
+	accountAddress = newAddr.String()
+}
+
+func (s *IntegrationTestSuite) TearDownSuite() {
+	s.T().Log("tearing down integration test suite")
+	s.network.Cleanup()
+}
+
+func CreateBond(s *IntegrationTestSuite) {
+	val := s.network.Validators[0]
+	sr := s.Require()
+
+	testCases := []struct {
+		name string
+		args []string
+		err  bool
+	}{
+		{
+			"create bond",
+			[]string{
+				fmt.Sprintf("100000000000%s", s.cfg.BondDenom),
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, accountName),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, fmt.Sprintf("3%s", s.cfg.BondDenom)),
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(fmt.Sprintf("Case %s", tc.name), func() {
```

Comment:
Test cases for `NewCreateBondCmd` already exist in the bond module cli testutil. No need to run test cases here.

Simply create a bond and assert that there is no error in the process.